packages/components/src/navigator/types.ts

```diff
@@ -26,6 +26,7 @@ export type NavigatorLocation = NavigateOptions & {
 export type Navigator = {
 	location: NavigatorLocation;
 	params: MatchParams;
+	hasBack: boolean;
```

Comment:
This is the simplest API I could think of. Happy to go with an alternative though. Some ideas:

- `historyLength: number` 
- Turn `goBack()` into `goBack(): boolean` which returns `true` if it actually went back